File: src/main/java/io/github/onepieceapi/services/impl/ChapterServiceImpl.java
package io.github.onepieceapi.services.impl;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import io.github.onepieceapi.model.Chapter;
import io.github.onepieceapi.repositories.ChapterRepository;
import io.github.onepieceapi.services.ChapterService;
import lombok.AllArgsConstructor;

@Service
@AllArgsConstructor
public class ChapterServiceImpl implements ChapterService {
    private final ChapterRepository chapterRepository;

    @Override
    public Page<Chapter> getAll(Pageable pageable) {
        return chapterRepository.findAll(pageable);
    }
}

File: src/main/kotlin/io/github/onePieceApi/services/impl/ChapterServiceImpl.kt
package io.github.onePieceApi.services.impl

import io.github.onePieceApi.model.Chapter
import io.github.onePieceApi.repositories.ChapterRepository
import io.github.onePieceApi.services.ChapterService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class ChapterServiceImpl constructor(
    private val chapterRepository: ChapterRepository
) : ChapterService {

    override fun getAll(pageable: Pageable): Page<Chapter> {
        return chapterRepository.findAll(pageable)
    }

}
